front/pages/w/[wId]/workspace/index.tsx

```diff
@@ -318,30 +318,47 @@ function SlackBotToggle({
         hasSeparatorIfLast={true}
         action={
           <div className="flex flex-row items-center gap-2">
-            {isSlackBotEnabled && (
+            {isSlackBotEnabled && slackBotDataSource && (
               <Button
                 variant="outline"
                 label="Reconnect"
                 size="xs"
                 icon={ArrowPathIcon}
                 onClick={async () => {
-                  const connectionIdRes = await setupConnection({
```

Comment:
We can't use setupConnection as it pins the oauth provider on the connector provider but here we have the slack provider but the slack_bot connector provider

Reply:
Not valid anymore signature added to support useCase